### ISSUE TITLE
ClusterRoleBindings doesn't get created when no subject is set

### ISSUE THREAD

**Issue by joschi36** [FairwindsOps/rbac-manager#79]
In my cluster no ClusterRoleBindings are getting created by rbac-manager when no subject is set.
Also no error is outputted.

**RBAC Definition (redacted)**
```yaml
- name: XXX-admins
  subjects:
    - kind: User
      name: 'XXX/john.doe@example.com'
  clusterRoleBindings:
    - clusterRole: cluster-admin
- name: YYY-admins
  subjects:
    - kind: User
      name: 'XXX/jane.doe@example.com'
    - kind: User
      name: 'XXX/sterling.archer@example.com'
  clusterRoleBindings:
    - clusterRole: XXX:customer-admin
  roleBindings:
    - clusterRole: admin
      namespaceSelector:
        matchExpressions:
          - key: customer-system
            operator: NotIn
            values:
              - "true"
- name: ZZZ-users
  subjects: []
  roleBindings:
    - clusterRole: admin
      namespaceSelector:
        matchExpressions:
          - key: customer-system
            operator: NotIn
            values:
              - "true"
```

**Role Bindings**
```
default       XXX-users-YYY-admins-admin   ClusterRole/admin      XXX/jane.doe@example.com, XXX/sterling.archer@example.com
default       XXX-users-YYY-users-admin    ClusterRole/admin                                                               
kube-public   XXX-users-YYY-admins-admin   ClusterRole/admin      XXX/jane.doe@example.com, XXX/sterling.archer@example.com
kube-public   XXX-users-YYY-users-admin    ClusterRole/admin                                                               
test          XXX-users-YYY-admins-admin   ClusterRole/admin      XXX/jane.doe@example.com, XXX/sterling.archer@example.com
test          XXX-users-YYY-users-admin    ClusterRole/admin                                                               
```

**Cluster Role Bindings**
No ClusterRoleBindings getting created

**Log**
```
time="2019-09-16T15:45:59Z" level=info msg=----------------------------------
time="2019-09-16T15:45:59Z" level=info msg="rbac-manager 0.8.2 running"
time="2019-09-16T15:45:59Z" level=info msg=----------------------------------
time="2019-09-16T15:45:59Z" level=info msg="Registering components"
time="2019-09-16T15:45:59Z" level=info msg="Watching resources related to RBAC Definitions"
time="2019-09-16T15:45:59Z" level=info msg="Watching RBAC Definitions"
time="2019-09-16T15:45:59Z" level=info msg="Reconciling RBACDefinition XXX-users"
time="2019-09-16T15:45:59Z" level=info msg="Reconciling default namespace for XXX-users"
time="2019-09-16T15:45:59Z" level=info msg="Reconciling kube-public namespace for XXX-users"
time="2019-09-16T15:45:59Z" level=info msg="Reconciling kube-system namespace for XXX-users"
time="2019-09-16T15:45:59Z" level=info msg="Reconciling monitoring namespace for XXX-users"
time="2019-09-16T15:46:00Z" level=info msg="Reconciling rbac-manager namespace for XXX-users"
time="2019-09-16T15:46:00Z" level=info msg="Reconciling test namespace for XXX-users"
time="2019-09-16T15:46:00Z" level=info msg="Reconciling cert-manager namespace for XXX-users"
```

**Comment by sudermanjr** [comment]
@joschi36 Thanks for submitting this issue. 

What behavior are you expecting when you don't set a subject?  I am thinking that perhaps an error or a warning should be emitted.

**Comment by joschi36** [comment]
@sudermanjr 
The problem is it doesn‘t create any clusterrolebindings. Not even for the other definitions.

**Comment by sudermanjr** [comment]
> @sudermanjr
> The problem is it doesn‘t create any clusterrolebindings. Not even for the other definitions.

I'm confused.  I see several bindings listed in the rbac-lookup output you posted in your original issue.  Could you please post the full rbac-lookup output and your expected rbac-lookup output?

**Comment by sudermanjr** [comment]
Ah, I see.  I think I can reproduce this now.  I will take a look.

**Comment by svenmueller** [comment]
We have the same issue when defining rolebindings with empty subjects (`[]`).

**Comment by sudermanjr** [comment]
This has been resolved by #323
